api/filters.go

```diff
@@ -93,7 +94,8 @@ func (api *API) createFilter(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := api.isValidDatasetDimensions(ctx, v, req.Dimensions, req.PopulationType); err != nil {
+	// update here for the first
+	if err := api.isValidDatasetDimensions(ctx, v, req.Dimensions, req.PopulationType, v.IsBasedOn.Type); err != nil {
```

Comment:
You seem to be passing two different 'types' here, dataset type in one place and filter type another, I don't think they don't hold the same set of values.. where a filter would be 'flexible'  a dataset would be of type 'cantabular_flexible_table'.